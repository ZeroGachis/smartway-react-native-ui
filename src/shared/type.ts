export type WithTestID<Props> = Props & { testID?: string };
