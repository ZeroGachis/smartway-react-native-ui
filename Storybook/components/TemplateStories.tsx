import type { Meta, StoryObj } from '@storybook/react-native';
import React from 'react';
import { StyleSheet, View } from 'react-native';

// fake component must be replace by one from the library
const Component = (props: React.PropsWithChildren) => {
    const { children } = props;
    return <>{children}</>;
}; // must be removed

type ComponentProps = React.ComponentProps<typeof Component>;

export default {
    title: 'components/Component', // change it with your component name eg('components/AwesomeButton',)
    component: Component,
    //default value
    args: {},
    // add some control in argTypes
    // some example
    // https://github.com/storybookjs/react-native/blob/next/examples/expo-example/components/ControlExamples/ControlExample/ControlExample.stories.tsx
    argTypes: {},
    // use it to wrap your storie in composant
    decorators: [
        (Story) => {
            const styles = StyleSheet.create({
                container: { paddingTop: 16 },
            });
            return (
                <View style={styles.container}>
                    <Story />
                </View>
            );
        },
    ],
} as Meta<ComponentProps>;

type Story = StoryObj<ComponentProps>;

// first story use it with args to change load only props value
// you can also use render as
// render : (args) => <AwesomeButton {...args}>
export const Default: Story = {
    args: {},
};
Default.parameters = { noSafeArea: false };
