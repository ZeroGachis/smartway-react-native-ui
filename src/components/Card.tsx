import type { ReactNode } from "react";
import React from "react";
import { StyleProp, View, ViewStyle } from "react-native";
import { useTheme } from "smartway-react-native-ui";

interface Props {
    children?: ReactNode,
    style: ViewStyle
};

export const Card = ({children, style}: Props) => {
    const theme = useTheme();
    const cardStyle: StyleProp<ViewStyle> = {
        backgroundColor: theme.sw.colors.neutral[50],
        width: '100%',
        alignItems: 'center',
        borderRadius: 12,
        paddingTop: theme.sw.spacing.xl,
        paddingBottom: theme.sw.spacing.xl,
        paddingLeft: theme.sw.spacing.m,
        paddingRight: theme.sw.spacing.m,
        ...style
        // TODO: Implement shadows
    };

    return (
        <View style={cardStyle}>
            {children}
        </View>
    );
};
