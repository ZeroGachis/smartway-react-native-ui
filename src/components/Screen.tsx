import React from 'react';
import { SafeAreaView, ViewStyle, StatusBar, StyleSheet } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { useTheme } from '../styles/themes';
import DeviceInfo from 'react-native-device-info';
import type { WithTestID } from 'src/shared/type';

type Props = WithTestID<{
    children?: React.ReactNode;
    style?: ViewStyle;
    statusBarColor?: string;
}>;

export const Screen = ({ children, style, testID, statusBarColor }: Props) => {
    const theme = useTheme();
    const insets = useSafeAreaInsets();
    const isTablet = DeviceInfo.isTablet();

    const styles = StyleSheet.create({
        screen: {
            flex: 1,
            backgroundColor: theme.sw.color.neutral[0],
            marginTop: insets.top,
            paddingLeft: isTablet ? theme.sw.spacing.s : 0,
            paddingRight: isTablet ? theme.sw.spacing.s : 0,
            ...style,
        },
    });

    return (
        <SafeAreaView style={styles.screen} testID={testID}>
            <StatusBar
                translucent
                backgroundColor={
                    statusBarColor
                        ? statusBarColor
                        : theme.sw.color.neutral[0]
                }
                barStyle='dark-content'
            />
            {children}
        </SafeAreaView>
    );
};
