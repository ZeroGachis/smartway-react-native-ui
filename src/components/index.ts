import { Button } from 'react-native-paper';
import Logo from './logos/logos';

export { Button, Logo };