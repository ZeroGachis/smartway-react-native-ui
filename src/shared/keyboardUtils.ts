import { useEffect } from 'react';
import { Keyboard, KeyboardEventListener, TextInput } from 'react-native';

export function useListenerOnKeyboardHiding(listener: KeyboardEventListener) {
    useEffect(() => {
        const hideSubscription = Keyboard.addListener(
            'keyboardDidHide',
            listener,
        );
        return () => {
            hideSubscription.remove();
        };
    }, [listener]);
}

export function hideKeyboard(ref: React.RefObject<TextInput>) {
    ref?.current?.blur();
}
