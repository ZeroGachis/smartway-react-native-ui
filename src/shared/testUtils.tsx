import React, { ReactElement } from 'react';
import {
    RenderOptions,
    act,
    render as rtlRender,
} from '@testing-library/react-native';
import { ThemeProvider } from '../styles/themes'; // Replace with the actual path to your ThemeProvider

const uiRender = (ui: ReactElement, options?: RenderOptions) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
        <ThemeProvider>{children}</ThemeProvider>
    );

    return rtlRender(ui, { wrapper: Wrapper, ...options });
};

const setupFakeTimer = () => {
    jest.useFakeTimers();

    act(() => {
        jest.runAllTimers();
    });
};

const cleanUpFakeTimer = () => {
    jest.clearAllTimers();
    jest.useRealTimers();
};

export * from '@testing-library/react-native';
export { uiRender as render, setupFakeTimer, cleanUpFakeTimer };
