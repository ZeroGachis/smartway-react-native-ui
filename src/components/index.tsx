import { Button } from './buttons/Button';
import { IconButton } from './buttons/IconButton';
import { Logo } from './logos/Logo';
import { Screen } from './Screen';
import { Headline } from './typography/Headline';
import { Body } from './typography/Body';
import { Modal } from './modals/Modal';
import { TextField } from './fields/TextField';
import { Icon } from './icons/Icon';
import { TopAppBar } from './topAppBar/TopAppBar';
import { Card } from './card/Card';
import { Tag } from './tag/Tag';
import { Divider } from './divider/Divider';
import { LineChart } from './charts/LineChart';
import { Badge } from './badge/Badge';
import { DateField } from './dateSelector/DateField';
import { DateSelector } from './dateSelector/DateSelector';
import { Label } from './typography/Label';
import { FieldState, NumberField } from './numberField/NumberField';

export { default as Snackbar, useSnackbar } from './alert/Snackbar';
export { default as Banner, useBanner } from './alert/Banner';

export {
    Body,
    Button,
    IconButton,
    Modal,
    Headline,
    Icon,
    Logo,
    Label,
    Screen,
    TextField,
    TopAppBar,
    NumberField,
    FieldState,
    Card,
    Tag,
    Divider,
    LineChart,
    Badge,
    DateField,
    DateSelector,
};
