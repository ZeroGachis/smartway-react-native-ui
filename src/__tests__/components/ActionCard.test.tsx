import React from 'react';
import { act, fireEvent, render } from '../../shared/testUtils';
import { ActionCard } from '../../components';
import { Text, View } from 'react-native';

const mockedCallback = jest.fn();
const mockedTestID = 'mockedTestID';
describe('MODULE | ActionCard', () => {
    it('component renders correctly', () => {
        const tree = render(
            <ActionCard
                titleColor={'black'}
                bottomChildren={
                    <View>
                        <Text>Test</Text>
                    </View>
                }
            >
                {'Test children'}
            </ActionCard>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('component renders correctly in disabled state', () => {
        const tree = render(
            <ActionCard
                disabled
                titleColor={'black'}
                bottomChildren={
                    <View>
                        <Text>Test</Text>
                    </View>
                }
            >
                {'Test children'}
            </ActionCard>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('fires callback on button press', async () => {
        const tree = render(
            <ActionCard
                onClear={mockedCallback}
                displayClear={true}
                titleColor={'black'}
                bottomChildren={
                    <View>
                        <Text>Test</Text>
                    </View>
                }
                buttonTestID={mockedTestID}
            >
                {'Test children'}
            </ActionCard>,
        );

        await act(async () => {
            return fireEvent.press(tree.getByTestId(mockedTestID));
        }),
            expect(mockedCallback).toHaveBeenCalled();
    });
});
