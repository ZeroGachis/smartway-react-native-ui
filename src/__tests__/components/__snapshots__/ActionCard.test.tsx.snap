// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MODULE | ActionCard component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#F4F6F8",
          "borderRadius": 20,
          "borderWidth": 1,
          "elevation": 2,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "PublicSans-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 19,
            "paddingTop": 16,
            "textAlign": "center",
          }
        }
      />
      <View
        style={
          {
            "padding": 16,
          }
        }
      />
      <View
        style={
          {
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#F4F6F8",
              "height": 1,
              "marginVertical": 32,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "paddingBottom": 32,
            }
          }
        >
          <View>
            <Text>
              Test
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`MODULE | ActionCard component renders correctly in disabled state 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F9FAFB",
          "borderColor": "#E5EAEE",
          "borderRadius": 20,
          "borderWidth": 1,
          "elevation": 2,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "PublicSans-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 19,
            "paddingTop": 16,
            "textAlign": "center",
          }
        }
      />
      <View
        style={
          {
            "padding": 16,
          }
        }
      />
      <View
        style={
          {
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#F4F6F8",
              "height": 1,
              "marginVertical": 32,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "paddingBottom": 16,
            }
          }
        >
          <View>
            <Text>
              Test
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
